src/internal/io.js

```diff
@@ -16,29 +16,14 @@ const FLUSH  = 'FLUSH'
 
 const effect = (type, payload) => ({[IO]: true, [type]: payload})
 
-export function take(channel, pattern) {
-  if(arguments.length >= 2) {
-    check(channel, is.notUndef, 'take(channel, pattern): channel is undefined')
-    check(channel, is.take, `take(channel, pattern): argument ${String(channel)} is not a valid channel (channel argument must have a take method)`)
-    check(pattern, is.notUndef, 'take(channel, pattern): pattern is undefined')
-    check(pattern, is.pattern, `take(channel, pattern): argument ${String(pattern)} is not a valid pattern (pattern must be String | Function: a => boolean | Array<String>)`)
-  } else if(arguments.length === 1) {
-    check(channel, is.notUndef, 'take(patternOrChannel): undefined argument')
-    if(!is.take(channel)) {
-      if(is.pattern(channel)) {
-        pattern = channel
-        channel = null
-      } else {
-        throw new Error(`take(patternOrChannel): argument ${String(channel)} is not valid channel or a valid pattern`)
-      }
-    } else {
-      pattern = '*'
-    }
-  } else {
-    pattern = '*'
+export function take(patternOrChannel = '*') {
```

Comment:
but w've still left the `patternOrChannel = '*'`. which will assign a '*' to the argument if it's undefined.

The actual code will transpile to 

```js
function take() {
  var patternOrChannel = arguments.length <= 0 || arguments[0] === undefined ? '*' : arguments[0];

  if (arguments.length) {
    check(patternOrChannel, is.notUndef, 'take(patternOrChannel): patternOrChannel is undefined');
  }
}
```

The default value will be assigned before our check has a chance to run